bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,212 @@
+        """Place multiple flags on the board"""
+        board: GameBoard = self.games[ctx.author.id]["revealed"]
+        for x, y in cords:
+            if board[y][x] == "hidden":
+                board[y][x] = "flag"
+
+        await self.reload_board(ctx)
+
+    async def lost(self, ctx: commands.Context) -> None:
+        """The player lost the game"""
+        game = self.games[ctx.author.id]
+        game["revealed"] = game["board"]
+        await self.reload_board(ctx)
+        await ctx.author.send(":fire: You lost :fire: ")
+        await game["chat_msg"].channel.send(f":fire: {ctx.author.mention} just lost minesweeper :fire:")
+        del self.games[ctx.author.id]
+
+    async def won(self, ctx: commands.Context) -> None:
+        """The player won the game"""
+        game = self.games[ctx.author.id]
+        game["revealed"] = game["board"]
+        await self.reload_board(ctx)
+        await ctx.author.send(":tada:  You won! :tada: ")
+        await game["chat_msg"].channel.send(f":tada: {ctx.author.mention} just won minesweeper :tada:")
+        del self.games[ctx.author.id]
+
+    def reveal_zeros(self, revealed: GameBoard, board: GameBoard, x: int, y: int) -> None:
+        """Used when a 0 is encountered to do a flood fill"""
+        for x_ in [x - 1, x, x + 1]:
+            for y_ in [y - 1, y, y + 1]:
+                if x_ == -1 or x_ == 10 or y_ == -1 or y_ == 10 or revealed[y_][x_] != "hidden":
+                    continue
+                revealed[y_][x_] = board[y_][x_]
+                if board[y_][x_] == 0:
+                    self.reveal_zeros(revealed, board, x_, y_)
+
+    async def check_if_won(self, ctx, revealed: GameBoard, board: GameBoard) -> bool:
+        """Checks if a player has won"""
+        for x_ in range(10):
+            for y_ in range(10):
+                if revealed[y_][x_] == "hidden" and board[y_][x_] != "bomb":
+                    return True
+        else:
+            await self.won(ctx)
+            return False
+
+    async def reveal_one(self, ctx: commands.Context, revealed: GameBoard, board: GameBoard, x: int, y: int) -> bool:
+        """Reveal one square."""
+        revealed[y][x] = board[y][x]
+        if board[y][x] == "bomb":
+            await self.lost(ctx)
+            return False
+        elif board[y][x] == 0:
+            self.reveal_zeros(revealed, board, x, y)
+        return await self.check_if_won(ctx, revealed, board)
+
+    @commands.dm_only()
+    @commands.command(name="reveal")
+    async def reveal_command(self, ctx: commands.Context, *cords: CordConverter) -> None:
+        """Reveal multiple cells"""
+        game = self.games[ctx.author.id]
+        revealed: GameBoard = game["revealed"]
+        board: GameBoard = game["board"]
+
+        reload_board = True
+        for x, y in cords:
+            if not await self.reveal_one(ctx, revealed, board, x, y):
+                reload_board = False
```

Comment:
What is this for? From what I can tell this only is set to `False` when  a player wins or loses. However, `won()` and `lost()` call `reload_board()` anyway so why not let this function handle revealing the board for all cases? The return values that accommodate this current system are unintuitive (e.g. `check_if_won` returning `False` when won) and difficult to follow.